models/repo/repo.go

```diff
@@ -290,7 +290,17 @@ func (repo *Repository) LoadUnits(ctx context.Context) (err error) {
 	}
 
 	repo.Units, err = getUnitsByRepoID(db.GetEngine(ctx), repo.ID)
-	log.Trace("repo.Units: %-+v", repo.Units)
+	if log.GetLevel() >= log.TRACE {
+		unitsStr := "["
+		for _, unit := range repo.Units {
+			unitsStr += unit.Type.String()
+			unitsStr += ", "
+		}
+		unitsStr = strings.TrimSuffix(unitsStr, ", ")
+		unitsStr += "]"
+		log.Trace("repo.Units: %s", unitsStr)
```

Comment:
Why `log.GetLevel() >= log.TRACE`? TRACE=0 while INFO=2.

If log.GetLevel() == INFO, should the log.Trace be executed?

And the code could be rewritten to `strings.Join`, or just use string formatter:

```
unitTypeStrings := make([]string, len(repo.Units))
for _, unit := range repo.Units {
    unitTypeStrings[i] = unit.Type.String()
}
log.Trace("repo.Units, ID=%d, Types: %v", repo.ID, unitTypeStrings)
```

Reply:
And there is a `IsTrace`:

![image](https://user-images.githubusercontent.com/2114189/155888591-46490b8a-a725-46e6-9e0c-9153fcb914d7.png)